openapi_python_client/templates/property_templates/union_property.py.jinja

```diff
@@ -17,24 +53,17 @@ def _parse_{{ property.python_name }}(data: object) -> {{ property.get_type_stri
         {% endif %}
     {% if inner_template.check_type_for_construct and (not loop.last or ns.contains_unmodified_properties) %}
     try:
-        if not {{ inner_template.check_type_for_construct(inner_property, "data") }}:
-            raise TypeError()
-        {{ inner_template.construct(inner_property, "data") | indent(8) }}
-        return {{ inner_property.python_name }}
+{{ construct_inner_property(inner_property) | indent(8, True) }}
```

Comment:
Even though you can see the logic here is slightly different from old lines 27-32 (it doesn't have the if/endif that are on old lines 27 and 30), I'm using the same macro for convenience. That works fine because we already know at this point, due to the test on old line 18, that `inner_template.check_type_for_construct` is true.